extensions/neo4j/deployment/src/main/java/io/quarkus/neo4j/deployment/Neo4jDriverProcessor.java

```diff
@@ -33,11 +35,13 @@ AdditionalBeanBuildItem createDriverProducer() {
 
     @BuildStep
     @Record(ExecutionTime.RUNTIME_INIT)
-    void configureDriverProducer(Neo4jDriverRecorder recorder, BeanContainerBuildItem beanContainerBuildItem,
+    Neo4jDriverBuildItem configureDriverProducer(Neo4jDriverRecorder recorder, BeanContainerBuildItem beanContainerBuildItem,
             Neo4jConfiguration configuration,
             ShutdownContextBuildItem shutdownContext) {
 
-        recorder.configureNeo4jProducer(beanContainerBuildItem.getValue(), configuration, shutdownContext);
+        Driver driver = recorder.initializeDriver(configuration, shutdownContext);
```

Comment:
I don't think that will work, you need to use a `RuntimeValue<Driver>`. We have plenty of examples if you look for `RuntimeValue` in the code.

Reply:
Yes, this is absolutely correct. It _might_ work even without it, but there is no point in risking it and not using `RuntimeValue`

Reply:
Just to clarify why this worked without `RuntimeValue`: It worked because `Driver` is an interface and as such the bytecode recorder knew to just create a proxy for it.

If however the class was not an interface, the whole endeavor would have failed 😉

Reply:
Thanks for getting back with this info. Might come in handy some time to know this!

Reply:
You are welcome!